### PR TITLE
Remove deprecated code from oneDPL documentation

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we make the next things:
- remove deprecated SYCL code from oneDPL documentation;
- unify order of included headers in documentations examples.

### CODE REVIEW

File: documentation/library_guide/parallel_api/execution_policies.rst [hunk 1]
```diff
@@ -117,17 +117,17 @@ and ``using namespace sycl;`` directives when referring to policy classes and fu
 
 .. code:: cpp
 
-   auto policy_a = device_policy<class PolicyA> {};
+   auto policy_a = device_policy<class PolicyA>();
    std::for_each(policy_a, ...);
 
 .. code:: cpp
 
-  auto policy_b = device_policy<class PolicyB> {device{gpu_selector{}}};
```
Comment (SergeyKopienko) [inline documentation/library_guide/parallel_api/execution_policies.rst:125]: Compile error in C++

File: documentation/library_guide/parallel_api/execution_policies.rst [hunk 2]
```diff
@@ -117,17 +117,17 @@ and ``using namespace sycl;`` directives when referring to policy classes and fu
...
+  auto policy_b = device_policy<class PolicyB>(device(gpu_selector_v));
   std::for_each(policy_b, ...);
 
 .. code:: cpp
 
-  auto policy_c = device_policy<class PolicyС> {cpu_selector{}};
```
Comment (SergeyKopienko) [inline documentation/library_guide/parallel_api/execution_policies.rst:130]: Incorrect codepage of lerrer ```С``` in ```PolicyС```; C++ compile error

File: documentation/library_guide/introduction/onedpl_gsg.rst [hunk 3]
```diff
@@ -128,7 +128,7 @@ fit into the square *S*. Random number generation is performed in scalar manner
 
     float estimated_pi;
     {
-        sycl::queue q(sycl::gpu_selector{});
```
Comment (SergeyKopienko) [inline documentation/library_guide/introduction/onedpl_gsg.rst:131]: gpu_selector is deprecated: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-selector

File: documentation/library_guide/parallel_api/execution_policies.rst [hunk 4]
```diff
@@ -117,17 +117,17 @@ and ``using namespace sycl;`` directives when referring to policy classes and fu
 
 .. code:: cpp
 
-   auto policy_a = device_policy<class PolicyA> {};
+   auto policy_a = device_policy<class PolicyA>();
```
Comment (akukanov) [inline documentation/library_guide/parallel_api/execution_policies.rst:120]: I would rather keep curly braces for variable initializations, as it clearly distinguishes these from function declarations and calls.

Comment (SergeyKopienko) [reply documentation/library_guide/parallel_api/execution_policies.rst:120]: Fixed

File: documentation/library_guide/onedpl_gsg.rst [hunk 5]
```diff
@@ -131,7 +131,7 @@ fit into the square *S*. Random number generation is performed in scalar manner
 
     float estimated_pi;
     {
-        sycl::queue q(sycl::gpu_selector{});
```
Comment (SergeyKopienko) [inline documentation/library_guide/onedpl_gsg.rst:134]: ```gpu_selector``` is deprecated: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-selector

File: documentation/library_guide/parallel_api/execution_policies.rst [hunk 6]
```diff
@@ -122,12 +122,12 @@ and ``using namespace sycl;`` directives when referring to policy classes and fu
 
 .. code:: cpp
 
-  auto policy_b = device_policy<class PolicyB> {device{gpu_selector{}}};
```
Comment (SergeyKopienko) [inline documentation/library_guide/parallel_api/execution_policies.rst:125]: ```gpu_selector``` is deprecated: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-selector

File: documentation/library_guide/parallel_api/execution_policies.rst (hunk 2 above)
Comment (SergeyKopienko) [reply documentation/library_guide/parallel_api/execution_policies.rst:130]: ```cpu_selector``` is deprecated: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:device-selector

File: documentation/library_guide/api_for_sycl_kernels/random.rst [hunk 7]
```diff
@@ -88,10 +88,10 @@ Random number generation is available for SYCL* device-side and host-side code.
 
 .. code:: cpp
 
+    #include <oneapi/dpl/random>
+    #include <sycl/sycl.hpp>
     #include <iostream>
```
Comment (SergeyKopienko) [inline documentation/library_guide/api_for_sycl_kernels/random.rst:93]: As I understand, usually the order of include headers assumed as:
1) oneDPL public headers
2) SYCL headers
3) STL headers

**Review by akukanov — APPROVED** [review]

**Review by dcbenito — APPROVED** [review]
Changes LGTM.

**Pull request merged**